### PR TITLE
[#ICC-121] Add message-status handler with port and adapter

### PR DESCRIPTION
**Pull request by fabriziopapi**
#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: outbound/adapter/queue-outbound-publisher.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      TE.tryCatch(
+        () =>
+          producer.sendMessage(
+            Buffer.from(JSON.stringify(document)).toString("base64")
+          ),
+        E.toError
+      ),
+      TE.mapLeft(error => {
+        throw error;
+      }),
```
Comment (gquadrati) [inline outbound/adapter/queue-outbound-publisher.ts:19]: would it be more reusable to move the throw outside this method?

File: AnalyticsMessageStatusFromChangeFeed/function.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      "connectionStringSetting": "COSMOS_API_CONNECTION_STRING",
+      "databaseName": "%COSMOSDB_NAME%",
+      "collectionName": "message-status",
+      "leaseCollectionPrefix": "%MESSAGE_STATUS_LEASES_PREFIX%-",
+      "leaseCollectionName": "pdnd-leases",
+      "createLeaseCollectionIfNotExists": true,
+      "startFromBeginning": true
+    }
+  ],
+  "scriptFile": "../dist/AnalyticsMessageStatusFromChangeFeed/index.js"
```
Comment (gquadrati) [inline AnalyticsMessageStatusFromChangeFeed/function.json:17]: We should add the `retry` block (I know it was missing, but it's needed)

File: businesslogic/analytics-message-status.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      RA.map(messageStatus =>
+        pipe(
+          mainPublisher.publish(messageStatus),
+          TE.orElse(_error => fallbackPublisher.publish(messageStatus)),
+          TE.toUnion,
+          T.map(constVoid)
+        )
+      )
+    ),
+    TT.fold(T.sequenceArray, T.sequenceArray, (e, r) =>
```
Comment (gquadrati) [inline businesslogic/analytics-message-status.ts:44]: Can we rename `e` and `r` for readability?

**Review by gquadrati — CHANGES_REQUESTED** [review]

File: outbound/adapter/queue-outbound-publisher.ts (hunk 1 above)
Comment (fabriziopapi) [reply outbound/adapter/queue-outbound-publisher.ts]: you are right, I've moved it outside.

File: AnalyticsMessageStatusFromChangeFeed/function.json (hunk 2 above)
Comment (fabriziopapi) [reply AnalyticsMessageStatusFromChangeFeed/function.json:23]: You are right, done!

File: businesslogic/analytics-message-status.ts (hunk 3 above)
Comment (fabriziopapi) [reply businesslogic/analytics-message-status.ts]: Renamed them!

**Pull request closed without merging**